### ISSUE TITLE
[nvim none-ls] No ESLint configuration found

### ISSUE THREAD

**Issue by piersolenski** [mantoni/eslint_d.js#311]
I've recently switched a project to the new `eslint.config.js` format. When using `eslint`, it works perfectly (albeit slow!), when I switch to `eslint_d` I get the following error:

```
Error: No ESLint configuration found in /Users/...
```

I am using the following version
```
eslint_d: v14.0.3, bundled eslint: v9.8.0
```

The tickets below suggest that it should be working with this version:
- https://github.com/mantoni/eslint_d.js/issues/281
- https://github.com/mantoni/eslint_d.js/issues/214 

I'm using `eslint_d` with Neovim and https://github.com/nvimtools/none-ls.nvim. My config looks like this:

```lua
-- Diagnostics, code actions, formatting, etc
return {
	"nvimtools/none-ls.nvim",
	lazy = false,
	dependencies = {
		"gbprod/none-ls-shellcheck.nvim",
		"jay-babu/mason-null-ls.nvim",
		"nvimtools/none-ls-extras.nvim",
		"williamboman/mason.nvim",
	},
	config = function()
		local null_ls = require("null-ls")
		local mason_null_ls = require("mason-null-ls")

		local formatting = null_ls.builtins.formatting
		local diagnostics = null_ls.builtins.diagnostics

		-- Helper to conditionally register eslint handlers only if eslint is
		-- config. If eslint is not configured for a project, it just fails.
		local function has_eslint_config(utils)
			return utils.root_has_file({
				".eslintrc",
				".eslintrc.cjs",
				".eslintrc.js",
				".eslintrc.json",
				"eslint.config.cjs",
				"eslint.config.js",
				"eslint.config.mjs",
			})
		end

		null_ls.setup({
			sources = {
				diagnostics.pylint,
				diagnostics.zsh,
				diagnostics.zsh,
				formatting.black,
				formatting.markdownlint,
				formatting.prettierd,
				formatting.shfmt,
				formatting.stylua,
				require("none-ls-shellcheck.code_actions"),
				require("none-ls-shellcheck.diagnostics"),
				require("none-ls.code_actions.eslint_d").with({ condition = has_eslint_config }),
				require("none-ls.diagnostics.eslint_d").with({ condition = has_eslint_config }),
				require("none-ls.formatting.eslint_d").with({ condition = has_eslint_config }),
				require("none-ls.formatting.jq"),
				require("none-ls.formatting.jq"),
			},
		})

		mason_null_ls.setup({
			ensure_installed = nil,
			automatic_installation = true,
			automatic_setup = false,
		})
	end,
}
```
(When switching `eslint_d` to `eslint` in the above config, everything works!)

Thanks for the awesome plugin!

**Comment by mantoni** [comment]
> `eslint_d: v14.0.3, bundled eslint: v9.8.0`

This means that no local eslint was resolved and it falls back to the bundled version. Is there something wrong with the current working directory?

Also, do you have the same problem on the command line when running `eslint_d <some-file>`?

**Comment by piersolenski** [comment]
It does indeed seem to work with the command line, I guess the problem is coming via my Neovim setup, although it was working before 9, and doesn't include anything crazy?

I've also noticed if I change my eslint config to include an incorrect setting, using `eslint` via `none-ls` will offer a suggestion of how to fix it:

```Oops! Something went wrong! :(ESLint: 9.9.0Configuration for rule "semi" is invalid. Expected severity of "off", 0, "warn", 1, "error", or 2.You passed '"never"'.See https://eslint.org/docs/latest/use/configure/rules#using-configuration-files for configuring rules. 1:1:14 eslint ```

`eslint_d` via `none-ls` just crashes:
```
failed to decode json: Expected value but found invalid token at character 1 1:1:3 eslint_d 
```

For anyone else experiencing issues with Neovim, for now I've switched to the eslint lsp and it seems pretty speedy:

```
require 'lspconfig'.eslint.setup({
  settings = {
    packageManager = 'yarn'
  },
  on_attach = function(client, bufnr)
    vim.api.nvim_create_autocmd("BufWritePre", {
      buffer = bufnr,
      command = "EslintFixAll",
    })
  end,
})
```

**Comment by L2jLiga** [comment]
Hi there, could you share your nvim start command and what is cwd?

I'm also using latest eslint_d via none-ls, but without any issues

**Comment by piersolenski** [comment]
It was simply `nvim` and whatever project I was working in.

**Comment by L2jLiga** [comment]
This is minimal config which is working fine on my end
https://github.com/L2jLiga/nvim-none-ls-eslint_d

Tested with Windows 11 23H2 and Fedora 40

**Comment by mantoni** [comment]
@piersolenski Can this be closed? Or do you think there's something wrong with `eslint_d`?

If you're still experiencing issues, there's a new version with `--debug` support (see README). Maybe that helps finding the cause.

**Comment by piersolenski** [comment]
Hey @mantoni, I've switched to using the eslint lsp. There were multiple points of potential failure with my convoluted neovim setup, and I can't be sure it was definitely `eslint_d`, so feel free to close this - especially if no one else if having this issue.

**Comment by L2jLiga** [comment]
> occur same error..

This comment is useless, if you have similar issue then:
1. Create new issue
2. Provide information about how you have it configured
3. Attach you nvim configuration or at least part related to eslint_d configuration
4. Attach logs if possible
5. Run eslid_d with `--debug` flag to see what exactly happening on eslint_d side

**Comment by bramdelta** [comment]
Had the same error. Ultimately boiled down to using a monorepo setup. If I started nvim from my `frontend/` folder it would format just fine on save, but if I did it from the root it would throw the error. Only after installing eslint in the root of the project would it not attempt to find the `.eslint_d` config file from within Mason (where I installed it,  this also occurred simliarly if I installed eslint_d via pnpm and not Mason). 

I do have eslint installed globally via pnpm, which is why I could run eslint from the root normally and it would work (the eslint.config.js in the root would read would simply import the one within `frontend/`). So, I suppose it's something to do with eslint_d being started from where it's binary lives and only "seeing" an eslint if node_modules (from nvim's CWD) has it installed, ignoring the globally installed eslint? Confusingly it still implies to be using the bundled eslint in both the case where my root has eslint installed in node_modules and when it didn't:

```
:!eslint_d --version
eslint_d: v15.0.2, bundled eslint: v10.1.0
```
